app/client/jsx/PuckSelect.jsx

```diff
@@ -8,107 +8,99 @@ class PuckSelect extends Component {
       }
 
   onButtonClick() { this.setState({ columnOpen: true }) }
-  onButtonClickReset() {
+  onButtonClickReset() { // to fix
     this.setState({ columnOpen: false })
     this.setState({ rowOpen: null })
     this.setState({ avatar: null })
-    this.forceUpdate()
+    // this.forceUpdate()
   }
-  handleClick(id) { this.setState({ rowOpen: id }) }
+  handleClickDesign(id) { this.setState({ rowOpen: id }) }
   handleClickColor(id) {
     this.setState({ avatar: id })
     this.props.handleSelect([this.state.rowOpen,id])
   }
 
   render() {
 
-    //reselect button WIP
-    if (this.state.avatar!=null) {
-      return (
-        <div className='outer'>
-          <input className='fade' type="button" onClick={this.onButtonClick} value="Pick your puck"/>
-          <div className='spacer'/>
-          <div><img className='image' src={avatars[this.state.rowOpen][this.state.avatar]}/></div>
-        </div>
-      )
-    } else {
+    if (this.state.avatar===null) { // checks whether selection not yet complete
 
+      // puck list generated from 0th colorway of each design
       let puck_list = avatars.map(function(arr,i) { return arr[0] }).slice(0, avatars.length - 1)
-      let puck_array
+      let all_avatar_designs_array
       let fade = this.props.opacity
-      if (this.state.columnOpen) {
+
+      if (this.state.columnOpen) { // checks whether selection begun by clicking button, opening column
 
         fade = 'fade'
+        all_avatar_designs_array = puck_list.map((puck,id) => {
 
-        puck_array = puck_list.map((puck,id) => {
+          if (this.state.rowOpen!==id) { // checks each item in column for whether selected
 
-          if (this.state.rowOpen===id) {
+            let handleClick = () => this.handleClickDesign(id)
+            let image_transparency = 'image'
+            if (this.state.rowOpen!=null && this.state.rowOpen!==id) {
+              image_transparency = 'non-selected-image'
+            }
+            return ( // returns single instance of each nonselected design
+              <PuckBox
+              handleClick={handleClick}
+              key={id}
+              image={puck}
+              style='box'
+              imageStyle={image_transparency}
+              />
+            )
 
-            let variant_array = avatars[this.state.rowOpen].map((color,id) => {
+          } else { // selected design
 
+            let colorway_variants_array = avatars[this.state.rowOpen].map((color,id) => {
               let selected = 'image'
-              // if (id===0) { selected = 'box-selected' }
-              let handleClickColor = () => this.handleClickColor(id)
-
-              // avatar color variant selection
-              return (
+              let handleClick = () => this.handleClickColor(id)
+              return ( // returns all colorways for selected design
                 <PuckBox
-                handleClick={handleClickColor}
+                handleClick={handleClick}
```

Comment:
It doesn't look like you modify selected, colorway, or handleClick, so these can be `const`s. Or you could just directly pass them to PuckBox. 

Same comment for the `handleClick` above